prime-router/src/main/kotlin/history/db/DeliveryDatabaseAccess.kt

```diff
@@ -172,7 +172,8 @@ class DeliveryDatabaseAccess(val db: DatabaseAccess = BaseEngine.databaseAccessS
         if (sentReportIdsForReceiver.isEmpty()) {
```

Comment:
(optional) With `sentReportIdsForReceiver` being used only to determine empty or not, would it be worth changing `fetchReportIdsForReceiverAndTask` to something that returns a boolean and use that instead? The underlying query can be changed to fetch a boolean if that simplification is beneficial from a performance perspective, or the "is empty or not" logic can just be offloaded to the function.

Reply:
Oops, didn't see Jack's comment, seems like we're talking about the same thing, feel free to disregard.